src/ui/features/referral-program/Invite.tsx

```diff
@@ -0,0 +1,485 @@
+import { getBackOrHome } from 'src/ui/shared/navigation/getBackOrHome';
+import { HStack } from 'src/ui/ui-kit/HStack';
+import CardsLeftSrc from 'src/ui/assets/cards-left.png';
+import CardsLeft2xSrc from 'src/ui/assets/cards-left@2x.png';
+import CardsRightSrc from 'src/ui/assets/cards-right.png';
+import CardsRight2xSrc from 'src/ui/assets/cards-right@2x.png';
+import ArrowDownIcon from 'jsx:src/ui/assets/caret-down-filled.svg';
+import AccountIcon from 'jsx:src/ui/assets/account.svg';
+import XpIcon from 'jsx:src/ui/assets/xp.svg';
+import ConnectIcon from 'jsx:src/ui/assets/technology-connect.svg';
+import QrCodeIcon from 'jsx:src/ui/assets/qr-code.svg';
+import type { WalletGroup } from 'src/shared/types/WalletGroup';
+import type { ExternallyOwnedAccount } from 'src/shared/types/ExternallyOwnedAccount';
+import type { BareWallet } from 'src/shared/types/BareWallet';
+import { FillView } from 'src/ui/components/FillView';
+import type { DeviceAccount } from 'src/shared/types/Device';
+import type { HTMLDialogElementInterface } from 'src/ui/ui-kit/ModalDialogs/HTMLDialogElementInterface';
+import { useWalletGroups } from 'src/ui/shared/requests/useWalletGroups';
+import { walletPort } from 'src/ui/shared/channels';
+import { WalletAvatar } from 'src/ui/components/WalletAvatar';
+import { WalletDisplayName } from 'src/ui/components/WalletDisplayName';
+import { useBackgroundKind } from 'src/ui/components/Background';
+import { PageColumn } from 'src/ui/components/PageColumn';
+import { PageTop } from 'src/ui/components/PageTop';
+import { Frame } from 'src/ui/ui-kit/Frame';
+import { Button } from 'src/ui/ui-kit/Button';
+import { BottomSheetDialog } from 'src/ui/ui-kit/ModalDialogs/BottomSheetDialog';
+import { CopyButton } from 'src/ui/components/CopyButton';
+import { DialogTitle } from 'src/ui/ui-kit/ModalDialogs/DialogTitle';
+import { normalizeAddress } from 'src/shared/normalizeAddress';
+import { KeyboardShortcut } from 'src/ui/components/KeyboardShortcut';
+import { PageBottom } from 'src/ui/components/PageBottom';
+import { CenteredDialog } from 'src/ui/ui-kit/ModalDialogs/CenteredDialog';
+import { invariant } from 'src/shared/invariant';
+import { isReadonlyContainer } from 'src/shared/types/validators';
+import { UnstyledAnchor } from 'src/ui/ui-kit/UnstyledAnchor';
+import { WalletList } from 'src/ui/pages/WalletSelect/WalletList';
+import type { ReferrerData } from 'src/modules/zerion-api/requests/check-referral';
+import { ReferralLinkDialog } from './ReferralLinkDialog';
+import { EnterReferralCodeDialog } from './EnterReferralCodeDialog';
+import { QRCodeDialog } from './QRCodeDialog';
+import { SuccessDialog } from './SuccessDialog';
+import { useWalletsMeta } from './shared/useWalletsMeta';
+import { ReferrerLink } from './shared/ReferrerLink';
+import * as styles from './styles.module.css';
+
+// TODO: Remove when explore rewards page is ready
+const SHOW_EXPLORE_REWARDS_BUTTON = process.env.NODE_ENV === 'development';
```

Comment:
Should this be a `FEATURE_LOYALTY_FLOW` flag?

Reply:
or is this something else? does this require web app?

Reply:
Not really. Users should never visit this page if the referral program is disabled in Firebase. The button should always be displayed if the referral program is enabled. I was just waiting for https://app.zerion.io/rewards to be ready. I've removed this constant ✅